force-app/main/default/classes/ContactTriggerExpensesService.cls

```diff
@@ -0,0 +1,34 @@
+public with sharing class ContactTriggerExpensesService {
+    public static List<Account> CheckingIsTheirExpensesChange(List<Contact> newTrigger,Map<ID, Contact> oldTrigger,Map<ID, Account> allAccounts){
+         System.debug('CheckingIsTheirExpensesChange');
+        List<Contact> contactForNextStep=new List<Contact>();
+        for(Contact contactNew:newTrigger){
+            if(contactNew.Expenses__c!=oldTrigger.get(contactNew.id).Expenses__c){
+                System.debug('CheckingIsTheirExpensesChange if reached');
+                contactForNextStep.add(contactNew);
+            }
+        }
+        return preparingAccountsForUpdateAfterExpensesChange(allAccounts, contactForNextStep);
```

Comment:
Nie róbmy tego w taki sposób, metody muszą wykonywać jedną operację, u Ciebie metodka nazwya się jak by sprawdzała coś i poprawnie pierwsza część wykonuje sprawdzenie czy expense się zmienił, ale w retrunie nie powinno się wywoływać następnej metody, zwłaszcza, że ta metoda w nazwie na to nie wskazuje. Więc tam wyżęj gdzie masz wywołanie tej metody, możesz tutaj zwrócić liste Contactów, na których Expense się zmienił i linijke niżej wywołać to preparingAccounts.... a tutaj w returnie zostawić tylko List<Contact> a metode możesz nazwać zamiast CheckingIsTheirExpensesChange to filterContactsWhereExpenseChange(); Dzięki temu wcale tutaj jeszcze nie potrzebujesz ID accountów